datafusion/sql/src/expr/arrow_cast.rs

```diff
@@ -580,8 +632,15 @@ mod test {
             DataType::Timestamp(TimeUnit::Millisecond, None),
             DataType::Timestamp(TimeUnit::Microsecond, None),
             DataType::Timestamp(TimeUnit::Nanosecond, None),
-            // TODO support timezones
-            //DataType::Timestamp(TimeUnit::Nanosecond, Some("UTC".into())),
+            // we can't cover all possible timezones, here we only test utc and +08:00
+            DataType::Timestamp(TimeUnit::Nanosecond, Some("+00:00".into())),
+            DataType::Timestamp(TimeUnit::Microsecond, Some("+00:00".into())),
+            DataType::Timestamp(TimeUnit::Millisecond, Some("+00:00".into())),
+            DataType::Timestamp(TimeUnit::Second, Some("+00:00".into())),
+            DataType::Timestamp(TimeUnit::Nanosecond, Some("+08:00".into())),
+            DataType::Timestamp(TimeUnit::Microsecond, Some("+08:00".into())),
+            DataType::Timestamp(TimeUnit::Millisecond, Some("+08:00".into())),
+            DataType::Timestamp(TimeUnit::Second, Some("+08:00".into())),
```

Comment:
Some other test cases that would be good:

`""` (empty timezone)

Also errors:

Three double quotes
```
"Timestamp(Nanosecond, Some("+00:00""))" 
```

Reply:
thanks @alamb , more error handling and test cases added